### PR TITLE
Refactor the dependency code to reduce duplication

### PR DESCRIPTION
**Pull request by bcrochet**
Simple refactor of the dependency (task and pipeline) code to reduce the
amount of duplicated code.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Review by jomkz — APPROVED** [review]
/lgtm

**Review by acornett21 — CHANGES_REQUESTED** [review]

File: controllers/dependencies.go [hunk 1]
```diff
@@ -77,26 +79,23 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(pipeline *cer
...
-
-	if err = yamlutil.Unmarshal(b, pipeline); err != nil {
+	if err = yamlutil.Unmarshal(b, &obj); err != nil {
 		log.Error(err, fmt.Sprintf("Couldn't unmarshall yaml file for: %s", fileName))
 		return err
 	}
 
-	pipeline.SetNamespace(obj.GetNamespace())
-	err = r.Get(context.Background(), types.NamespacedName{Name: pipeline.Name, Namespace: pipeline.Namespace}, pipeline)
+	obj.SetNamespace(owner.GetNamespace())
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply controllers/dependencies.go:94]: Ok. I'll fix and make sure it reads properly.

File: controllers/dependencies.go [hunk 2]
```diff
@@ -77,26 +79,23 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(ctx context.C
...
-	if err = yamlutil.Unmarshal(b, pipeline); err != nil {
+	if err = yamlutil.Unmarshal(b, &obj); err != nil {
 		log.Error(err, fmt.Sprintf("Couldn't unmarshall yaml file for: %s", fileName))
 		return err
 	}
 
-	pipeline.SetNamespace(obj.GetNamespace())
-	err = r.Get(ctx, types.NamespacedName{Name: pipeline.Name, Namespace: pipeline.Namespace}, pipeline)
+	obj.SetNamespace(owner.GetNamespace())
+	err = r.Client.Get(context.Background(), types.NamespacedName{Name: owner.GetName(), Namespace: owner.GetNamespace()}, obj)
```
[the remark this answers is not in the document]
Comment (bcrochet) [reply controllers/dependencies.go]: Good catch. Fixed.

File: controllers/dependencies.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (acornett21) [reply controllers/dependencies.go:94]: I think this is fine, I just had to test the code to see which line was wrong 94/95. It's harder to review code you've wrote. :)

**Review by acornett21 — APPROVED** [review]

**Pull request merged**